pkg/app/js_nowasm.go

```diff
@@ -215,6 +215,13 @@ func (w *browserWindow) addHistory(u *url.URL) {
 func (w *browserWindow) replaceHistory(u *url.URL) {
 }
 
+// JSValue returns the underlying syscall/js value of the given Javascript
+// value.
+// Returns the parameter v for non-wasm builds.
+func JSValue(v Value) any {
```

Comment:
Why do we need to add this in non wasm file?

Reply:
Similar to `app.ValueOf`:
https://github.com/maxence-charriere/go-app/blob/4af747695016174dfbe4502d0754da7ea1b4807b/pkg/app/js.go#L168-L170
There is a js implementation:
https://github.com/maxence-charriere/go-app/blob/4af747695016174dfbe4502d0754da7ea1b4807b/pkg/app/js_wasm.go#L146-L149
and a nowasm implementation:
https://github.com/maxence-charriere/go-app/blob/4af747695016174dfbe4502d0754da7ea1b4807b/pkg/app/js_nowasm.go#L156-L158

The nowasm code path is only there to allow compiles to work for both js and nowasm builds.

Reply:
The question is more what is the usecase of this? The purpose is to deal with syscall/js which is never usable out of js env. this is the very reason why go-app wrapping it.

Reply:
Apologies for the lack of explanation on this :smile: 

`app.JSValue` returns the underlying `syscall/js.Value` when provided an `app.Value` :heavy_check_mark: 

Since this function signature returns `syscall/js.Value` it can only be compiled for `GOOS=js`

For issue #876 there is a need to return a promise to a javascript call in a UI component:

```go
promise := app.Window().Get("Promise").New(app.FuncOf(goFunc))
```
Returning the promise directly will fail since `app.Value` cannot be converted by `syscall/js.ValueOf`
```go
// fails with panic: ValueOf: invalid value
return promise
```
Using `app.JSValue` works, but the component will only compile for `GOOS=js`
```go
return app.JSValue(promise)
```

With https://github.com/maxence-charriere/go-app/pull/907 and the removal of `app.jsval` this may be challenging to solve this use case.

Let me know if you'd like to see a PR to address this.